src/node_trace_events.cc

```diff
@@ -0,0 +1,172 @@
+#include "node_internals.h"
+#include "tracing/agent.h"
+
+namespace node {
+
+using v8::Context;
+using v8::FunctionCallbackInfo;
+using v8::Int32;
+using v8::Local;
+using v8::Object;
+using v8::Value;
+
+extern tracing::Agent* tracing_agent;
+
+// The tracing APIs require category groups to be pointers to long-lived
+// strings. Those strings are stored here.
```

Comment:
Looking at the V8 source code, this doesn’t seem to be true? (https://github.com/nodejs/node/blob/master/deps/v8/src/libplatform/tracing/tracing-controller.cc#L168 & related places use string comparisons, not pointer comparisons)

Reply:
Yeah, I'm confused too. This is where it is documented: https://github.com/nodejs/node/blob/master/src/tracing/trace_event_common.h#L123

Reply:
Hm, okay … /cc @matthewloring

Reply:
I really can’t make out a code path where this is actually needed … but it would be really good to have that verified by somebody who is familiar with the tracing code

Reply:
Agreed. It is very confusing. I think `trace_events` runs in a separate thread, can that affect it somehow?